os/binfmt/binfmt_exit.c

```diff
@@ -107,6 +111,19 @@ int binfmt_exit(FAR struct binary_s *bin)
 #endif
 	DEBUGASSERT(bin != NULL);
 
+#ifdef CONFIG_DEBUG_BIN_SYMBOL
+	bin_addr_info_t *info;
+	info = (bin_addr_info_t *)sq_peek(&g_bin_addr_list);
+	while (info) {
+		if (strncmp(info->bin_name, bin->bin_name, BIN_NAME_MAX) == 0) {
+			sq_rem((sq_entry_t *)info, &g_bin_addr_list);
+			kmm_free(info);
+			break;
+		}
+		info = (FAR void *)sq_next((FAR sq_entry_t *)info);
+	}
+#endif
```

Comment:
g_bin_addr_list is global, so set null its head and tail explicitly, i mean sq_init.

Reply:
This is not the whole list termination, only for exiting one binary(app).